oxidation/libparsec/crates/types/src/error.rs

```diff
@@ -2,9 +2,21 @@
 
 use thiserror::Error;
 
-use crate::{DateTime, DeviceID, EntryID, RealmID, UserID};
+use crate::{DateTime, DeviceID, EntryID, RealmID, Regex, UserID};
 use libparsec_crypto::CryptoError;
 
+#[derive(Error, Debug)]
+pub enum RegexError {
+    #[error("Regex parsing err: {err}")]
+    ParseError { err: regex::Error },
+    #[error("Failed to convert glob pattern into regex: {err}")]
+    GlobPatternError { err: fnmatch_regex::error::Error },
+    #[error("Failed to open pattern file: {err}")]
```

Comment:
An IO error is not only about failure to open a file, can you change the message to be more generic ?